core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/MemoryStore.java

```diff
@@ -185,6 +189,23 @@ public long getSyncDelay() {
 	}
 
 	/**
+	 * @return Returns the {@link EvaluationStrategy}.
+	 */
+	public synchronized EvaluationStrategyFactory getEvaluationStrategyFactory() {
```

Comment:
Will having this method synchronised instead of using volatile double-checked locking affect concurrent read thread performance on MemoryStore? Similar for the recently added FederatedServiceResolver methods below that could also use volatile double-checked locking instead.